convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -0,0 +1,221 @@
+        loggerinst.debug("Setting RHEL kernel %s as the default boot loader entry." % output.strip())
+        output, ret_code = utils.run_subprocess(["/usr/sbin/grubby", "--set-default", output.strip()])
+        if ret_code:
+            loggerinst.warning("Couldn't set the default GRUB2 boot loader entry:\n%s" % output)
+            self.add_message(
+                level="WARNING",
+                id="UNABLE_TO_SET_GRUB2_BOOT_LOADER_ENTRY",
+                title="Unable to set the GRUB2 boot loader entry",
+                description="Couldn't set the default GRUB2 boot loader entry:\n%s" % output,
+            )
+
+
+class FixDefaultKernel(actions.Action):
+    id = "FIX_DEFAULT_KERNEL"
+    dependencies = ("FIX_INVALID_GRUB2_ENTRIES",)
+
+    def run(self):
+        """
+        Systems converted from Oracle Linux or CentOS Linux may have leftover kernel-uek or kernel-plus in
+        /etc/sysconfig/kernel as DEFAULTKERNEL.
+        This function fixes that by replacing the DEFAULTKERNEL setting from kernel-uek or kernel-plus to kernel for
+        RHEL7 and kernel-core for RHEL8.
+        """
+        super(FixDefaultKernel, self).run()
+
+        loggerinst = logging.getLogger(__name__)
+
+        loggerinst.info("Checking for incorrect boot kernel")
+        kernel_sys_cfg = utils.get_file_content("/etc/sysconfig/kernel")
+
+        possible_kernels = ["kernel-uek", "kernel-plus"]
+        kernel_to_change = next(
+            iter(kernel for kernel in possible_kernels if kernel in kernel_sys_cfg),
+            None,
+        )
+        if kernel_to_change:
+            loggerinst.warning("Detected leftover boot kernel, changing to RHEL kernel")
+            self.add_message(
+                level="WARNING",
+                id="LEFTOVER_BOOT_KERNEL_DETECTED",
+                title="Leftover boot kernel detected",
+                description="Detected leftover boot kernel, changing to RHEL kernel",
+            )
+            # need to change to "kernel" in rhel7 and "kernel-core" in rhel8
+            new_kernel_str = "DEFAULTKERNEL=" + ("kernel" if system_info.version.major == 7 else "kernel-core")
+
+            kernel_sys_cfg = kernel_sys_cfg.replace("DEFAULTKERNEL=" + kernel_to_change, new_kernel_str)
+            utils.store_content_to_file("/etc/sysconfig/kernel", kernel_sys_cfg)
+            loggerinst.info("Boot kernel %s was changed to %s" % (kernel_to_change, new_kernel_str))
+        else:
+            loggerinst.debug("Boot kernel validated.")
+
+
+class KernelPkgsInstall(actions.Action):
+    id = "KERNEL_PACKAGES_INSTALLATION"
+    dependencies = ("VERIFY_RHEL_KERNEL_INSTALLED",)
+
+    def run(self):
+        """Install kernel packages and remove non-RHEL kernels."""
+        super(KernelPkgsInstall, self).run()
+
+        kernel_pkgs_to_install = pkghandler.remove_non_rhel_kernels()
+        if kernel_pkgs_to_install:
+            pkghandler.install_additional_rhel_kernel_pkgs(kernel_pkgs_to_install)
```

Comment:
Same for those two functions. They are only being used here, so we could move them to this class or this module

Reply:
Sounds good